tests/test_workflow_pipeline.py

```diff
@@ -723,6 +724,21 @@ def test_check_pipeline_version(
     assert f"using version {expected_version}" in caplog.text
 
 
+@pytest.mark.parametrize(
+    "variables,valid", [({"var1": "val1"}, True), ({"var2": None}, False)]
+)
+def test_check_pipeline_variables(workflow: PipelineWorkflow, variables, valid):
+    workflow.config.PIPELINE_VARIABLES.PROCESSING[workflow.pipeline_name][
+        workflow.pipeline_version
+    ] = variables
+    with (
+        pytest.raises(ValueError, match="Variable .* is not set in the config")
+        if not valid
+        else nullcontext()
```

Comment:
🍒 I find `nullcontext() if valid else pytest.raises(ValueError, match="Variable .* is not set in the config")` more readable, but this only a personal preference.